src/utils/instrument.hpp

```diff
@@ -0,0 +1,43 @@
+//========================================================================================
+// (C) (or copyright) 2023. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef UTILS_INSTRUMENT_HPP_
+#define UTILS_INSTRUMENT_HPP_
+
+#include <string>
+
+#include <Kokkos_Core.hpp>
+
+#define PARTHENON_INSTRUMENT KokkosTimer internal_inst_(__FILE__, __LINE__, __func__);
```

Comment:
```suggestion
#define PARTHENON_INSTRUMENT KokkosTimer internal_inst_func___LINE__(__FILE__, __LINE__, __func__);
```

Reply:
implemented this basic idea in a way that actually works ;)